modules/Bio/EnsEMBL/Compara/RunnableDB/HomologyAnnotation/DiamondBlastp.pm

```diff
@@ -0,0 +1,100 @@
+
+    my $start_member_id = $self->param_required('start_member_id');
+    my $end_member_id   = $self->param_required('end_member_id');
+
+    #Get list of members and sequences
+    my $member_ids = $self->compara_dba->get_HMMAnnotAdaptor->fetch_all_seqs_missing_annot_by_range($start_member_id, $end_member_id, 'no_null');
+    return $self->compara_dba->get_SeqMemberAdaptor->fetch_all_by_dbID_list($member_ids);
+}
+
+sub run {
+    my $self = shift @_;
+
+    #my $diamond_exe             = $self->param('diamond_exe'); 
+    # This will be in ENV.pm once installed properly in farm, currently diamond is installed locally in my $USER .bin/
+    my $diamond_exe             = 'diamond';
+    my $blast_params            = $self->param('blast_params')  || '';  # no parameters to C++ binary means having composition stats on and -seg masking off
+    my $evalue_limit            = $self->param('evalue_limit');
+    my $tophits                 = $self->param('tophits');
+
+    my $worker_temp_directory   = $self->worker_temp_directory;
+
+    my $blast_infile  = $worker_temp_directory . '/blast.in.'.$$;     # only for debugging
+    my $blast_outfile = $worker_temp_directory . '/blast.out.'.$$;    # looks like inevitable evil (tried many hairy alternatives and failed)
+
+    Bio::EnsEMBL::Compara::Utils::Preloader::load_all_sequences($self->compara_dba->get_SequenceAdaptor, undef, $self->param('query_set'));
+
+    if ($self->debug) {
+        print "diamond_infile $blast_infile\n";
+        my $members = $self->param('query_set')->get_all_Members;
+        foreach my $member ( @$members ) {
+            print Dumper $member unless $member->isa('Bio::EnsEMBL::Compara::SeqMember');
+        }
+        $self->param('query_set')->print_sequences_to_file($blast_infile, -format => 'fasta');
+    }
+
+    $self->compara_dba->dbc->disconnect_if_idle();
+
+    my $cross_pafs = [];
+    foreach my $blast_db (keys %{$self->param('all_blast_db')}) {
+        my $target_genome_db_id = $self->param('all_blast_db')->{$blast_db};
+
+        my $cmd = "$diamond_exe blastp -d $blast_db --evalue $evalue_limit --out $blast_outfile --outfmt 6 qseqid sseqid evalue score nident pident qstart qend sstart send length positive ppos qseq_gapped sseq_gapped $blast_params";
+
+        my $run_cmd = $self->write_to_command($cmd, sub {
+                my $blast_fh = shift;
+                $self->param('query_set')->print_sequences_to_file($blast_fh, -format => 'fasta');
+        } );
+        print "Time for diamond search " . $run_cmd->runtime_msec . " msec\n";
+
+        my $features = $self->parse_blast_table_into_paf($blast_outfile, $self->param('genome_db_id'), $target_genome_db_id);
+
+        unless($self->param('expected_members') == scalar(keys(%{$self->param('num_query_member')}))) {
+            # Most likely, this is happening due to MEMLIMIT, so make the job sleep if it parsed 0 sequences, to wait for MEMLIMIT to happen properly.
+            sleep(5);
```

Comment:
This is quite an interesting behaviour (and neat trick!).

Reply:
It was "borrowed" can't take credit